app/main.ipynb

```diff
@@ -570,18 +750,39 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 30,
-   "outputs": [],
    "source": [
-    "# write your numpy code here"
+    "def unknown_signature(lst, mask, new_value):\n",
+    "    lst_np = np.array(lst)\n",
+    "    mask_np = np.array(mask)\n",
+    "    lst_np[mask_np] = new_value\n",
+    "    last_updated = lst_np.tolist()\n",
+    "    return last_updated\n",
+    "\n",
+    "lst = [1, 2, 3, 4, 5]\n",
+    "mask = [True, False, True, False, True]\n",
+    "new_value = 0\n",
+    "unknown_signature(lst, mask, new_value)\n",
+    "lst\n",
```

Comment:
The original list `lst` is not updated in the NumPy version of the function. Consider returning the updated list and assigning it back to `lst`.